### DISCUSSION TITLE
Spock Framework Version Question

### DISCUSSION THREAD

**Discussion by jjathman** [renovatebot/renovate#17560] [Request Help] [answered]
We use the [Spock testing framework](https://spockframework.org) in our projects. It has a relatively unique naming scheme for artifacts that is causing us an issue. I'm not quite sure how to setup the Renovate configuration to avoid updates which we don't want.

we currently depend on version `org.spockframework:spock-core:2.1-groovy-3.0`, we would like to get an update to `org.spockframework:spock-core:2.2-groovy-3.0` however, Renovate is picking up `org.spockframework:spock-core:2.2-groovy-4.0` instead. This is for a version of Groovy we don't currently use.

Any suggestions on how to configure a package rule to only look for the `groovy-3.0` version?

**Comment by PhilipAbed** [comment, marked as answer]
Renovate will look into the datasource for all available version, and it will find that the first version it should change is that, so you need to restrict the versions.
i haven't tested this, but i'm sure you will get the idea, try something like this:
```
  "packageRules": [
    {
      "matchPackageNames": ["org.spockframework:spock-core"],
      "allowedVersions": "/^[0-9]+\\.[0-9]+(\\.[0-9]+)?-groovy-3\\.0$/"
    }
  ]
```

let me know if it works

**Reply by jjathman** [reply]
Thank you very much. It will be a few days before I can test it but I’ll let you know.

**Reply by jjathman** [reply]
This worked just as I expected. Thank you!

**Comment by mpern** [comment]
a more fancy(?) rule using `versionCompatibility`, in my case for `spock-bom`

```json
    {
      "matchDatasources": ["maven"],
      "matchPackageNames": ["org.spockframework:spock-bom"],
      "versionCompatibility": "^(?<version>[^-]+(?<compatibility>-.*))$"
    }
```
